File: tau WinAPP driver/CalculatorPython/notepad_test.py
import unittest
from appium import webdriver
from notepad import notepad_PO
from selenium.webdriver.common.keys import Keys

class notepad_test(unittest.TestCase):
    notepadSession = None

    def setUp(self):
        print("setup")
        desired_cap = {}
        desired_cap["app"] = "notepad.exe"
        self.notepadSession = webdriver.Remote(
            command_executor="http://127.0.0.1:4723",
            desired_capabilities=desired_cap
        )

    def tearDown(self):
        self.notepadSession.quit()

    
    def test_notepad(self):
        text = "awesome!!!"
        np = notepad_PO.notepad_PO(self.notepadSession)
        np.textArea().send_keys(text)

        np.textArea().send_keys(Keys.ALT,Keys.F4)
        np.dialogCancel().click()
        np.textArea().send_keys(Keys.ALT,Keys.F4)
        np.dialogDontSave().click()

File: tau WinAPP driver/CalculatorJava/src/test/java/Notepad_test.java
import io.appium.java_client.windows.WindowsDriver;
import org.aspectj.lang.annotation.After;
import org.openqa.selenium.Keys;
import org.openqa.selenium.remote.DesiredCapabilities;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

import java.net.MalformedURLException;
import java.net.URL;

public class Notepad_test {
    private WindowsDriver notepad = null;
    private Notepad_PO np = null;

    @BeforeClass
    public void setup(){
        System.out.println("setup");
        DesiredCapabilities capabilities = new DesiredCapabilities();
        capabilities.setCapability("app","notepad.exe");
        try{
            notepad = new WindowsDriver(new URL("http://127.0.0.1:4723"),capabilities);
            np =  new Notepad_PO(notepad);
        } catch (MalformedURLException e){
            e.printStackTrace();
        }

    }
    @AfterClass
    public void teardown(){
        System.out.println("teardown");
        notepad.quit();

    }

    @Test
    public void testNotepad(){
        System.out.println("Test Notepad");
        String text = "Windriver is awesome";
        np.TextArea().sendKeys(text);
        Assert.assertEquals(np.TextArea().getText(), text);


        np.TextArea().sendKeys(Keys.ALT,Keys.F4);
        np.DialogCancel().click();
        np.TextArea().sendKeys(Keys.ALT,Keys.F4);
        np.DialogDontSave().click();
    }

}
